python-packages/presto_drf/presto_drf/parsers.py

```diff
@@ -0,0 +1,65 @@
+import json
+
+from allianceutils.util.camel_case import camel_to_underscore
+from allianceutils.util.camel_case import underscoreize
+from django.conf import settings
+from rest_framework.exceptions import ParseError
+from rest_framework.parsers import JSONParser
+from rest_framework.parsers import MultiPartParser
+
+
+class CamelCaseJSONParser(JSONParser):
+    def parse(self, stream, media_type=None, parser_context=None):
+        parser_context = parser_context or {}
+        encoding = parser_context.get("encoding", settings.DEFAULT_CHARSET)
+
+        try:
+            data = stream.read().decode(encoding)
+            return underscoreize(json.loads(data))
+        except ValueError as exc:
+            raise ParseError(f"JSON parse error") from exc
+
+
+class CamelCaseMultiPartParser(MultiPartParser):
+    """
+    If we have received any files along with other json data it gets converted
+    on frontend into multi-part post with a single key for all JSON data (formData)
```

Comment:
Why does this happen? Is it not possible to treat it as a vanilla multipart form?

Reply:
No. Normal json data can still be nested so it comes through serialised under one field